beacon-chain/p2p/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package p2p
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/network"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p/encoder"
+)
+
+type Composite interface {
```

Comment:
Can this have a more descriptive name?

Reply:
Looking for suggestions here. Let me know what you think!

Reply:
maybe `P2P` ?

Reply:
`p2p.P2P` :joy:

Reply:
oh, nvm. Yeah we definitely dont want `p2p.P2P` lol

Reply:
That's what I went with! It's fine, just a bit goofy